### PR TITLE
LRA-678-lsa-ride-share-reservation-show-page-buttons-design

### PR DESCRIPTION
**Pull request by rsmoke**
I set up the reservation show page buttons (LRA-678) 
I used nested grids and created the layout described by Maria L's design. I did not use the class buttons that were pre-designed since these buttons required special treatment.

### CODE REVIEW

File: app/views/reservations/show.html.erb [hunk 1]
```diff
@@ -36,74 +36,89 @@
...
+          <% if (is_student?(current_user) && allow_student_to_edit_reservation?(@reservation) || is_manager?(current_user) && allow_manager_to_edit_reservation?(@reservation)) && !@reservation.approved %>
+            <div class="grid grid-cols-1 gap-4 place-content-center">
+              <% if recurring?(@reservation) %>
+                <%= form_tag cancel_recurring_reservation_url, method: :post, local: true,
+                  data: {
+                      approve_target: "form1"
+                    } do |form| %>
+                  <div class="my-2">
+                    <label for="cancel_type" class="fancy_label">Cancel Recurring Reservation</label>
+                    <%= select_tag :cancel_type, options_for_select(cancel_types), include_blank: "Select to Cancel ...", class: "filter_select",
```
Comment (britaumich) [inline app/views/reservations/show.html.erb:92]: Line 91 has a label to Cancel Recurring Reservation in the students' interface. You changed the same label for the admin interface to Cancel Reservations(s). And the Students Cancel button looks like this now:
Screenshot 2023-09-26 at 6 50 46 PM

I would keep the word 'recurring ' in this label to make things clear. This we can discuss, but the students's button should be edited anyway.

Comment (rsmoke) [reply app/views/reservations/show.html.erb:92]: I set the css and syntax to make the two recurring labels equal. 
I thought visually the 'reservation(s)' gave the same message as having the extra word 'recurring'. If you want the word recurring in the label then i will add it back.

Comment (britaumich) [reply app/views/reservations/show.html.erb:92]: The labels look good.
About the word 'recurring' in labels. I don't know what is better. There is a label 'Approve All Recurring Reservations'. I am working on 'Update All Recurring Reservations'. I am afraid we'll need something like 'Send All Recurring Update'. Then it makes sense to have 'Cancel Recurring Reservations'. 
Or maybe there should be a section 'Recurring' with all recurring buttons in there.
Should we ask Maria and others about all these buttons and labels?
I don't have a solution that I like.

Comment (rsmoke) [reply app/views/reservations/show.html.erb:92]: I would have Maria give us her input to the proper naming. I used the '... reservation(s)' to shorten the label

Comment (britaumich) [reply app/views/reservations/show.html.erb:92]: I'll show some screenshots tomorrow at the meeting including Update All Recurring Reservations bittons.

**Comment by britaumich** [comment]
I see that you changed the labels' and buttons' styles and instead of creating CSS classes and using them in the reservations show page you put styles into the page code.
Could you please create new classes for buttons and labels? I tried to merge your branch into mine with more buttons (updating all recurring reservations, etc.)  and had difficulty resolving the show page conflict.

**Comment by britaumich** [comment]
@rsmoke Please try to decrease the screen size - the buttons are not responding very well.

**Comment by rsmoke** [comment]
I set the break point to be 1 grid col at lg and less screen sizes

**Comment by britaumich** [comment]
@rsmoke this is what I see on a phone: 
Screenshot 2023-10-05 at 6 36 05 PM

**Comment by rsmoke** [comment]
Yup, looks good

**Comment by britaumich** [comment]
> Yup, looks good

What do you mean 'looks good'? What about the word Update between buttons?
Screenshot 2023-10-05 at 6 36 05 PM

**Comment by rsmoke** [comment]
I propose sending the whole button set to be designed better. Then send it
back to me for implementing.

-

Rick Smoke, Application Architect

For assistance please submit a request at:
https://teamdynamix.umich.edu/TDClient/47/LSAPortal/Requests/TicketRequests/NewForm?ID=lMiqIZLcXJ0_&RequestorType=ServiceOffering

College of Literature, Science, and the Arts | University of Michigan
LSA Technology Services | 301 East Liberty Street, Ann Arbor, MI 48109

**Comment by rsmoke** [comment]
I fixed the text overflow

**Comment by britaumich** [comment]
@rsmoke I think that on the mobile device buttons should be on the top of the screen not after the reservation fields. Right now they are on the top.
I think we have to ask the users' opinions about buttons. For me, the front is too small on new buttons. Also, it's easier to select the correct button when that are different sizes. When all of them are the same, I have to pay extra attention to select the correct one.

**Pull request closed without merging**